### PR TITLE
[CALCITE-2945] Use Boolean#equals in Boolean object compare

### PR DESCRIPTION
**Pull request by leesf**
Use Boolean#equals method in Boolean object compare in ConnectionPropertiesImpl#merge function.

### CODE REVIEW

**Comment by leesf** [comment]
cc @risdenk @joshelser

**Comment by risdenk** [comment]
FYI @leesf you don't need to cc myself or Josh on each PR. There are committers who see the JIRAs and PRs.

**Comment by leesf** [comment]
@risdenk Sorry to bother you.

File: core/src/main/java/org/apache/calcite/avatica/ConnectionPropertiesImpl.java [hunk 1]
```diff
@@ -94,11 +94,11 @@ public boolean isDirty() {
     if (this == that) {
       return this;
     }
-    if (that.isAutoCommit() != null && this.autoCommit != that.isAutoCommit()) {
+    if (that.isAutoCommit() != null && !this.autoCommit.equals(that.isAutoCommit())) {
       this.autoCommit = that.isAutoCommit();
       this.isDirty = true;
     }
-    if (that.isReadOnly() != null && this.readOnly != that.isReadOnly()) {
+    if (that.isReadOnly() != null && !this.readOnly.equals(that.isReadOnly())) {
```
Comment (danny0405) [inline core/src/main/java/org/apache/calcite/avatica/ConnectionPropertiesImpl.java]: +1, nice catch.

**Comment by danny0405** [comment]
@leesf It looks good, it would be better if you can supply a test case for the bad case when we use `==` as comparator.

**Comment by leesf** [comment]
@danny0405 Updated this pr to address your comment.

**Comment by julianhyde** [comment]
What about if that.readOnly is not null, but this.readOnly is null? The new code will throw NPE.

I suggest

```if (that.readOnly != null && !Objects.equals(this.readOnly, that.readOnly))```

The same bug exists in `autoCommit`.

We should use the same pattern in `transactionIsolation`, even though it doesn't have the bug.

In the test, please fix indentation, and use `assertThat( ... is())` rather than `assertEquals`.

**Comment by leesf** [comment]
@julianhyde Thanks for your reviewing, updated this pr to address your comment.

**Pull request closed without merging**